### PR TITLE
Utility to get object Path using location and inventory type

### PR DESCRIPTION
**Pull request by ArchanaKakani**
This function is added to support PCIe-Topology.
Util function takes Location code and Inventory Item Type
as inputs and return the dbus object path.

Tested:
  Reset Reload operation and normal poweon workrs fine.
  Also executed utility function with valid/invalid sets of input
  and got expected results.

Signed-off-by: ArchanaKakani <archana.kakani@ibm.com>

### CODE REVIEW

File: common/utils.cpp [hunk 1]
```diff
@@ -685,5 +706,37 @@ std::string getCurrentSystemTime()
     return ss.str();
 }
 
+std::string getLocationObject(std::string locationCode,
```
Comment (manojkiraneda) [inline common/utils.cpp:709]: this cannot be in generic code ..move this to oem.

This is a very specific use case for ibm (i dont think any one would have this kind of requirement)

File: common/utils.cpp [hunk 2]
```diff
@@ -390,6 +390,27 @@ PropertyValue DBusHandler::getDbusPropertyVariant(
...
+                                          "GetManagedObjects");
+        auto reply = bus.call(method);
+        reply.read(objects);
+    }
+    catch (const std::exception& e)
+    {
+        std::cerr << "Look up of objects failed : " <<  service << std::endl;
+    }
+
+    return objects;
```
Comment (manojkiraneda) [inline common/utils.cpp:411]: is the dbus call fails ..what should be returned ?

a util function should not really have exception handling - its the function that calls this API that needs exception handling

File: common/utils.cpp (hunk 1 above)
Comment (ArchanaKakani) [reply common/utils.cpp]: Moved this function to oem/ibm

File: common/utils.cpp (hunk 2 above)
Comment (ArchanaKakani) [reply common/utils.cpp:403]: Handled the exception in caller function

File: libpldmresponder/fru.cpp [hunk 3]
```diff
@@ -157,17 +157,12 @@ void FruImpl::buildFRUTable()
                          panelHotplugMatch);
 
     fru_parser::DBusLookupInfo dbusInfo;
-    // Read the all the inventory D-Bus objects
-    auto& bus = pldm::utils::DBusHandler::getBus();
+    dbusInfo = parser.inventoryLookup();
```
Comment (manojkiraneda) [inline libpldmresponder/fru.cpp:160]: why was this moved out of try ?

File: oem/ibm/libpldmresponder/utils.cpp [hunk 4]
```diff
@@ -501,6 +507,48 @@ void hostPCIETopologyIntf(
         "/xyz/openbmc_project/pldm", mctp_eid, hostEffecterParser);
 }
 
+std::string getLocationObject(std::string locationCode,
+                              std::string inventoryItemType)
```
Comment (manojkiraneda) [inline oem/ibm/libpldmresponder/utils.cpp:511]: std::string getObjectPathByLocationCode(const std::string& locationcode, const std::string& inventoryItemType)

File: oem/ibm/libpldmresponder/utils.cpp [hunk 5]
```diff
@@ -501,6 +507,48 @@ void hostPCIETopologyIntf(
...
+{
+    std::string path("");
```
Comment (manojkiraneda) [inline oem/ibm/libpldmresponder/utils.cpp:513]: the default constructor of std::string -> would construct an empty string.

File: libpldmresponder/fru.cpp (hunk 3 above)
Comment (ArchanaKakani) [reply libpldmresponder/fru.cpp]: Moved the function inside try block

File: oem/ibm/libpldmresponder/utils.cpp (hunk 4 above)
Comment (ArchanaKakani) [reply oem/ibm/libpldmresponder/utils.cpp]: Done

File: oem/ibm/libpldmresponder/utils.cpp (hunk 5 above)
Comment (ArchanaKakani) [reply oem/ibm/libpldmresponder/utils.cpp]: Done

**Review by manojkiraneda — APPROVED** [review]

**Review by sampmisr — APPROVED** [review]
